OpenSim/Common/DataTable.h

```diff
@@ -178,46 +178,175 @@ class AbstractDataTable {
         return _dependentsMetaData;
     }
 
-    /** Set metadata associated with the dependent columns.                   */
+    /** %Set metadata associated with the dependent columns.                  */
     void 
     setDependentsMetaData(const DependentsMetaData& dependentsMetaData) {
         _dependentsMetaData = dependentsMetaData;
         validateDependentsMetaData();
     }
 
-    /** Get column labels.                                                    */
+    /** Get column labels.                                                    
+
+    \throws KeyNotFound If column labels have not be set for the table.       */
     std::vector<std::string> getColumnLabels() const {
-        const auto& metadata = getDependentsMetaData();
-        const auto& absArray = metadata.getValueArrayForKey("labels");
+        const auto& absArray = 
+            _dependentsMetaData.getValueArrayForKey("labels");
         std::vector<std::string> labels{};
         for(size_t i = 0; i < absArray.size(); ++i)
             labels.push_back(absArray[i].getValue<std::string>());
 
         return labels;
     }
 
-    /** Set column labels.                                                    */
-    void setColumnLabels(const std::vector<std::string>& columnLabels) {
+    /** Get column label of a given column.                                   
+
+    \throws ColumnIndexOutOfRange If columnIndex is out of range of number of
+                                  columns.                                    
+    \throws KeyNotFound If column labels have not be set for the table.       */
+    const std::string& getColumnLabel(const size_t columnIndex) const {
```

Comment:
In this and the previous (and all column wise access) methods, It should be clearly documented that the columns indices, or iterate are only across the dependent columns of the table, with the independent column having its own accessors..